controllers/pkg/reconcilers/repository/reconciler_test.go

```diff
@@ -16,187 +16,260 @@ package repository
-			giteaClient: NephioGiteaClientMock{},
-			cr:          infrav1alpha1.Repository{},
-			expectedErr: nil,
-		},
-		"User Info reports error": {
-			ctx: nil,
-			giteaClient: NephioGiteaClientMock{
-				myUserInfoError: errors.New("Error getting User Information"),
+	tests := []repoTest{
+		{
+			name:   "User Info and Delete Repo both OK",
+			fields: fields{resource.NewAPIPatchingApplicator(nil), nil, nil, log.FromContext(context.Background())},
+			args: args{
+				nil,
+				nil,
+				&infrav1alpha1.Repository{
+					ObjectMeta: v1.ObjectMeta{
+						Name: "repo-name",
+					},
+				},
 			},
-			cr:          infrav1alpha1.Repository{},
-			expectedErr: errors.New("Error getting User Information"),
-		},
-		"Delete repo reports error": {
-			ctx: nil,
-			giteaClient: NephioGiteaClientMock{
-				deleteRepoError: errors.New("Error deleting repo"),
+			mocks: []mockHelper{
+				{"GetMyUserInfo", []string{}, []interface{}{&gitea.User{UserName: "gitea"}, nil, nil}},
+				{"DeleteRepo", []string{"string", "string"}, []interface{}{&gitea.Response{}, nil, nil}},
 			},
-			cr:          infrav1alpha1.Repository{},
-			expectedErr: errors.New("Error deleting repo"),
-		},
-	}
-	for tn, tc := range testCases {
-		t.Run(tn, func(t *testing.T) {
-			r := reconciler{}
-			err := r.deleteRepo(tc.ctx, tc.giteaClient, &tc.cr)
-			require.Equal(t, tc.expectedErr, err)
+			wantErr: false,
+		}, {
+			name:   "User Info reports error",
+			fields: fields{resource.NewAPIPatchingApplicator(nil), nil, nil, log.FromContext(context.Background())},
+			args: args{
+				nil,
+				nil,
+				&infrav1alpha1.Repository{
+					ObjectMeta: v1.ObjectMeta{
+						Name: "repo-name",
+					},
+				},
+			},
+			mocks: []mockHelper{
+				{"GetMyUserInfo", []string{}, []interface{}{&gitea.User{UserName: "gitea"}, nil, fmt.Errorf("Error getting User Information")}},
+			},
+			wantErr: true,
+		}, {
+			name:   "Delete Repo reports error",
+			fields: fields{resource.NewAPIPatchingApplicator(nil), nil, nil, log.FromContext(context.Background())},
+			args: args{
+				nil,
+				nil,
+				&infrav1alpha1.Repository{
+					ObjectMeta: v1.ObjectMeta{
+						Name: "repo-name",
+					},
+				},
+			},
+			mocks: []mockHelper{
+				{"GetMyUserInfo", []string{}, []interface{}{&gitea.User{UserName: "gitea"}, nil, nil}},
+				{"DeleteRepo", []string{"string", "string"}, []interface{}{&gitea.Response{}, fmt.Errorf("Error deleting repo")}},
+			},
+			wantErr: true,
+		}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &reconciler{
+				APIPatchingApplicator: tt.fields.APIPatchingApplicator,
+				giteaClient:           tt.fields.giteaClient,
+				finalizer:             tt.fields.finalizer,
+			}
+
+			initMockeryMocks(&tt)
+
+			if err := r.deleteRepo(tt.args.ctx, tt.args.giteaClient, tt.args.cr); (err != nil) != tt.wantErr {
+				t.Errorf("deleteRepo() error = %v, wantErr %v", err, tt.wantErr)
+			}
 		})
 	}
 }
+
+func initMockeryMocks(tt *repoTest) {
```

Comment:
Suggestion: This function could be moved to a the testing folder as well to be reused by others.

Reply:
Done.